.test-infra/jenkins/LoadTestConfig.groovy

```diff
@@ -0,0 +1,621 @@
+        private static class SourceOptions implements SerializableOption<String> {
+            private def _numRecords
+            private def _keySizeBytes
+            private def _valueSizeBytes
+            private def _numHotKeys
+            private def _hotKeyFraction
+            private def _splitPointFrequencyRecords
+
+            //internal usage
+            private SDK i_sdk
+
+            private SourceOptions() {}
+
+            static withSDK(final SDK sdk) {
+                final def input = new SourceOptions()
+                input.i_sdk = sdk
+                return input
+            }
+
+            void numRecords(final int num) { _numRecords = num }
+            void keySizeBytes(final int size) { _keySizeBytes = size }
+            void valueSizeBytes(final int size) { _valueSizeBytes = size }
+            void numHotsKeys(final int num) { _numHotKeys = num }
+            void hotKeyFraction(final int fraction) { _hotKeyFraction = fraction }
+            void splitPointFrequencyRecords(final int splitPoint) { _splitPointFrequencyRecords = splitPoint }
+
+            @Override
+            String toPrimitiveValues() {
+                return '\'' + new JsonBuilder(ConfigHelper.convertProperties(propertiesMap, i_sdk)).toString() + '\''
+            }
+
+            static SourceOptions of(final SourceOptions oldOptions) {
+                final def newOptions = new SourceOptions()
+                InvokerHelper.setProperties(newOptions, oldOptions.propertiesMap)
+                return newOptions
+            }
+
+            Map<String, Object> getPropertiesMap() {
+                return [
+                        i_sdk: i_sdk,
+                        _numRecords: _numRecords,
+                        _keySizeBytes: _keySizeBytes,
+                        _valueSizeBytes: _valueSizeBytes,
+                        _numHotKeys: _numHotKeys,
+                        _hotKeyFraction: _hotKeyFraction,
+                        _splitPointFrequencyRecords: _splitPointFrequencyRecords
+                ]
+            }
+        }
+
+        private static class StepOptions implements SerializableOption<String> {
+            private def _outputRecordsPerInputRecord
+            private boolean  _preservesInputKeyDistribution
+
+            //internal usage
+            private SDK i_sdk
+
+            private StepOptions() {}
+
+            static withSDK(final SDK sdk) {
+                final def option = new StepOptions()
+                option.i_sdk = sdk
+                return option
+            }
+
+            void outputRecordsPerInputRecord(final int records) { _outputRecordsPerInputRecord = records }
+            void preservesInputKeyDistribution(final boolean  shouldPreserve) { _preservesInputKeyDistribution = shouldPreserve }
+
+            @Override
+            String toPrimitiveValues() {
+                return '\'' + new JsonBuilder(ConfigHelper.convertProperties(propertiesMap, i_sdk)).toString() + '\''
```

Comment:
What do you think about changing it into notation "'${...}'"?

Reply:
Thanks, will change it!